prime-dotnet-webapi/Controllers/DocumentAccessController.cs

```diff
@@ -0,0 +1,85 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Prime.Auth;
+using Prime.Models.Api;
+using Prime.Services;
+using Prime.Services.Clients;
+
+namespace Prime.Controllers
+{
+    [Produces("application/json")]
+    [Route("api/document-access")]
+    [ApiController]
+    public class DocumentAccessController : ControllerBase
+    {
+        private readonly IDocumentService _documentService;
+        private readonly IDocumentAccessTokenService _documentAccessTokenService;
+        private readonly IDocumentManagerClient _documentManagerClient;
+
+        public DocumentAccessController(
+            IDocumentService documentService,
+            IDocumentAccessTokenService documentAccessTokenService,
+            IDocumentManagerClient documentManagerClient)
+        {
+            _documentService = documentService;
+            _documentAccessTokenService = documentAccessTokenService;
+            _documentManagerClient = documentManagerClient;
+        }
+
+        // GET: api/document-access/file-download/{accessTokenId}
+        /// <summary>
+        /// Gets the Document for downloading based on the supplied Document Access Token GUID.
+        /// </summary>
+        /// <remarks>This endpoint is not authenticated.</remarks>
+        [HttpGet("/file-download/{accessTokenId}", Name = nameof(GetDocumentByAccessToken))]
+        [AllowAnonymous]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public async Task<IActionResult> GetDocumentByAccessToken(Guid accessTokenId)
+        {
+            var documentAccessToken = await _documentAccessTokenService.GetDocumentAccessNoTrackingAsync(accessTokenId);
+
+            if (documentAccessToken == null)
+            {
+                return NotFound();
+            }
+
+            var response = await _documentManagerClient.GetFileAsync(documentAccessToken.DocumentGuid);
+
+            if (response == null)
+            {
+                return NotFound();
+            }
+
+            Response.Headers.Add("Content-Disposition", response.Content.Headers.ContentDisposition.ToString());
+
+            return File(await response.Content.ReadAsStreamAsync(), "application/octet-stream");
```

Comment:
Doesn't download immediately when the browser opens up.  Looking at force download, but doesn't prevent review since it works.